lib/std/net/test.zig

```diff
@@ -404,3 +404,58 @@ test "non-blocking tcp server" {
     const msg = buf[0..len];
     try testing.expect(mem.eql(u8, msg, "hello from server\n"));
 }
+
+test "non-blocking tcp client" {
+    if (builtin.os.tag == .wasi) return error.SkipZigTest;
+
+    if (builtin.os.tag == .windows) {
+        _ = try std.os.windows.WSAStartup(2, 2);
+    }
+    defer {
+        if (builtin.os.tag == .windows) {
+            std.os.windows.WSACleanup() catch unreachable;
+        }
+    }
+
+    // todo: Linux and BSD refuse the connection to the socket when it doesn't exist
+    if (builtin.os.tag == .windows) {
+        // attempt to connect to an address without a backing server to simulate pending connection
+
+        const address = std.net.Address.initIp4(.{ 127, 0, 0, 1 }, 999);
+
+        const tpe: u32 = std.posix.SOCK.STREAM | std.posix.SOCK.NONBLOCK | std.posix.SOCK.CLOEXEC;
+        const protocol = std.posix.IPPROTO.TCP;
+        const socket = try std.posix.socket(address.any.family, tpe, protocol);
+        defer std.posix.close(socket);
+
+        // first call returns .WouldBlock like usual
+        try testing.expectError(error.WouldBlock, std.posix.connect(socket, &address.any, address.getOsSockLen()));
+
+        // second call returns EALREADY or WSAEALREADY which is now translated to .ConnectionPending
+        try testing.expectError(error.ConnectionPending, std.posix.connect(socket, &address.any, address.getOsSockLen()));
```

Comment:
If the intention of this is to test that `EALREADY`/`WSAEISCONN` is mapped to `ConnectionPending`, but it can't reliably be reproduced cross-platform, I'd just omit this whole block from the test. It also seems potentially flaky since on Windows it still relies on a particular port not being connectable.